internal/venice-common/src/main/java/com/linkedin/venice/blob/ServerBlobFinder.java

```diff
@@ -0,0 +1,54 @@
+package com.linkedin.venice.blob;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.helix.HelixCustomizedViewOfflinePushRepository;
+import com.linkedin.venice.meta.Instance;
+import com.linkedin.venice.meta.Partition;
+import com.linkedin.venice.meta.Version;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class ServerBlobFinder implements BlobFinder {
+  private final HelixCustomizedViewOfflinePushRepository customizedViewRepository;
+
+  private static final Logger LOGGER = LogManager.getLogger(ServerBlobFinder.class);
+
+  public ServerBlobFinder(CompletableFuture<HelixCustomizedViewOfflinePushRepository> customizedViewRepository) {
+    this.customizedViewRepository = customizedViewRepository.join();
+  }
+
+  @Override
+  public BlobPeersDiscoveryResponse discoverBlobPeers(String storeName, int version, int partitionId) {
+    BlobPeersDiscoveryResponse response = new BlobPeersDiscoveryResponse();
+    try {
+      String currentVersionResource = Version.composeKafkaTopic(storeName, version);
+      // Get the partition assignments for the current version
+      List<String> hostNames = new ArrayList<>();
+      for (Partition partition: customizedViewRepository.getPartitionAssignments(currentVersionResource)
+          .getAllPartitions()) {
+        if (partition.getId() == partitionId) {
+          for (Instance instance: partition.getReadyToServeInstances()) {
+            String host = instance.getHost();
```

Comment:
@adamxchen Should we not specify the protocol port here somehow?  Otherwise we'd need to have some kind of client server agreement set in cfg2, which is a bit contrary to how discovery works in other contexts.

@sixpluszero?

Reply:
Our internal discussion leaves it outside of this interface. There will be a BlobManger thing that takes in this implementation, and there it is suppose to know port.
But I believe this client-server talking will be very much similar to II main-fork process. We will have to specify cfg for port instead of hardcode it somewhere. Both side should setup Netty server and client to talk to each other with the specific allocated port.

Reply:
Essentially, this interface does only one thing, look up the peers from different sources. If you think discovery is not a good name, we can adjust it later. I don't have a strong opinion in API naming, but just to mean that this API returns list of hostname having this specific resource.